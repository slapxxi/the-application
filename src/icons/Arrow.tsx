import * as React from 'react';

interface Props {
  size?: number;
  direction?: 'up' | 'down' | 'left' | 'right';
}

function Arrow({ size = 16, direction = 'down' }: Props) {
  const transform = directionToTransform(direction);
  return (
    <svg width={size} height={size} viewBox="0 0 100 100">
      <title>Arrow Icon</title>
      <path
        d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
        transform={transform}
      />
    </svg>
  );
}

function directionToTransform(direction) {
  switch (direction) {
    case 'up':
      return 'rotate(180, 50, 50)';
    case 'right':
      return 'rotate(-90, 50, 50)';
    case 'down':
      return undefined;
    case 'left':
      return 'rotate(90, 50, 50)';
    default:
      return undefined;
  }
}

export default Arrow;
