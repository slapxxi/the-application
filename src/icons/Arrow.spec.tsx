import React from 'react';
import { render } from 'react-testing-library';
import Arrow from './Arrow';

it('renders', () => {
  const { container } = render(<Arrow />);
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts "size" prop', () => {
  const { container } = render(<Arrow size={32} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts "up" direction', () => {
  const { container } = render(<Arrow direction="up" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts "right" direction', () => {
  const { container } = render(<Arrow direction="right" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts "down" direction', () => {
  const { container } = render(<Arrow direction="down" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('accepts "left" direction', () => {
  const { container } = render(<Arrow direction="left" />);
  expect(container.firstChild).toMatchSnapshot();
});
