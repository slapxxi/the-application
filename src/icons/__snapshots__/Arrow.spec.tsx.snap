// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts "down" direction 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
  />
</svg>
`;

exports[`accepts "left" direction 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
    transform="rotate(90, 50, 50)"
  />
</svg>
`;

exports[`accepts "right" direction 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
    transform="rotate(-90, 50, 50)"
  />
</svg>
`;

exports[`accepts "size" prop 1`] = `
<svg
  height="32"
  viewBox="0 0 100 100"
  width="32"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
  />
</svg>
`;

exports[`accepts "up" direction 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
    transform="rotate(180, 50, 50)"
  />
</svg>
`;

exports[`renders 1`] = `
<svg
  height="16"
  viewBox="0 0 100 100"
  width="16"
>
  <title>
    Arrow Icon
  </title>
  <path
    d="M50,81.73,3.51,35.24A8,8,0,0,1,14.82,23.93L50,59.1,85.17,23.93A8,8,0,1,1,96.49,35.24Z"
  />
</svg>
`;
